plugins/ipam/host-local/backend/allocator/allocator.go

```diff
@@ -215,6 +215,7 @@ func (i *RangeIter) Next() (*net.IPNet, net.IP) {
 
 	if i.startIP == nil {
 		i.startIP = i.cur
+		i.startRange = i.rangeIdx
 	} else if i.rangeIdx == i.startRange && i.cur.Equal(i.startIP) {
```

Comment:
Now that I look at this, do we even need `startRange`? Since we validate that ranges can't overlap, that should be sufficient, right?

Reply:
haha, good self-question :) , maybe I should walk all lines to figure this out

Reply:
@mars1024 any thoughts on whether we can remove `startRange`?

Reply:
yeah, I think `startRange` is redundant and I'll remove it